src/main/java/com/carrotcreative/recyclercore/widget/ProgressRecyclerViewLayout.java

```diff
@@ -112,4 +141,74 @@ public void scrollRecyclerViewToBottom()
             mRecyclerView.getLayoutManager().scrollToPosition(mRecyclerView.getAdapter().getItemCount()-1);
         }
     }
+
+    // =================================== //
+    // ========== Empty States =========== //
+    // =================================== //
+
+    /**
+     *
+     * @param emptyStateView The view that will be shown then the item count in the adapter is 0
+     */
+    public void setEmptyStateView(View emptyStateView)
+    {
+        /**
+         * If there is already an emplty state view, replace it.
+         */
+        if(mEmptyStateView != null)
+        {
+            mEmptyStateContainer.removeView(mEmptyStateView);
+        }
+        mEmptyStateView = emptyStateView;
+        mEmptyStateContainer.addView(emptyStateView);
+    }
+
+    /**
+     *
+     * @param visible true if you want to show the empty state, false other wise.
+     *                This will show the empty state only if the item count in adapter is 0.
+     *                If you want to skip item count, use #forceShowEmptyState()
+     */
+    public void setEmptyStateEnabled(boolean visible)
+    {
+        mEmptyStateVisible = visible;
+
+        /**
+         * If adapter not set do not show the empty state.
+         */
+        if(mAdapter == null || mAdapter.getItemCount() != 0 || visible == false)
+        {
+            if(mEmptyStateContainer.getVisibility() == VISIBLE)
+            {
+                mEmptyStateContainer.setVisibility(GONE);
+            }
+        }
+        else if(mAdapter.getItemCount() == 0)
+        {
+            showEmptyState();
+        }
+    }
+
+    /**
+     * The empty state is shown if the number of items in the adapter are 0.
+     * If you want to force show the Empty state, call this functions.
+     * Please keep in mind that when you set the Adapter, the empty state internal
+     * state is changed.
+     */
+    public void forceShowEmptyState()
```

Comment:
Technically `showEmptyState` is private, so it could not be called from outside of the class. 
I think when we call `ForceShowEmptyState` we should not set the `mEmptyStateVisible = true`